### ISSUE TITLE
Support Web Crypto API to enable async hashing

### ISSUE THREAD

**Issue by thernstig** [puleos/object-hash#123]
https://nodejs.org/api/webcrypto.html supports async hash functions. Meaning that if this libs implements an async function like:

```js
import hash from 'object-hash';

await hash({foo: 'bar'}) // => '67b69634f9880a282c14a0f0cb7ba20cf5d677e9'
```

Then users could use async functionality for this.

**Comment by strogonoff** [comment]
AFAICT this library relies on Node’s `.createHash()` exclusively, which is not supported even by browsers supporting `globalThis.crypto`.

**Comment by thernstig** [comment]
@strogonoff I am not sure I follow, would something like this not work? The intention is to replace `.createHash()` with the Web Crypto API.

```js
  const encode = new TextEncoder();
  const digest = await webcrypto.subtle.digest('SHA-512', encode.encode(str));
  return Buffer.from(digest).toString('hex');
```

**Comment by strogonoff** [comment]
@thernstig Ah yes. I was actually thinking along the same lines, that it’s possible to implement it in a way that works in both Node and browser. 

```typescript
// web version
new Uint8Array(await crypto.subtle.digest('SHA-1', (new TextEncoder()).encode('…'))).toString()
// Node version
new Uint8Array(await require('crypto').webcrypto.subtle.digest('SHA-1', (new TextEncoder()).encode('…'))).toString()
```

I saw that currently the library relies on Node’s flow, and I guess `browserify` fills in the gaps so that it works in browser (including browsers that may not support web crypto), that’s what I meant. But yes, it doesn’t have to be that way I guess.

**Comment by strogonoff** [comment]
(Note that `Buffer` is a Node-only construct, so an isomorphic solution would probably use `Uint8Array` instead.)

**Comment by thernstig** [comment]
Yes. So the main gist of using the Web Crypto API is that it supports `await` making it async. Second thing is of course that the Web Crypto API is probably the future for Node.js as well.

**Comment by strogonoff** [comment]
To me the reason for looking into this was not async but clearer and isomorphic logic. Currently there is browserify adding some unnecessary magic and increasing bundle size.

**Comment by ivosabev** [comment]
Are there any plans for this to be released?
